genesyscloud/architect_user_prompt/resource_genesyscloud_architect_user_prompt.go

```diff
@@ -25,14 +26,17 @@ func getAllUserPrompts(ctx context.Context, clientConfig *platformclientv2.Confi
 	resources := make(resourceExporter.ResourceIDMetaMap)
 	proxy := getArchitectUserPromptProxy(clientConfig)
 
-	userPrompts, resp, err := proxy.getAllArchitectUserPrompts(ctx, true, true, "")
-	if err != nil {
-		return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("failed to get user prompts: %s", err), resp)
-	}
-	for _, userPrompt := range *userPrompts {
-		resources[*userPrompt.Id] = &resourceExporter.ResourceMeta{Name: *userPrompt.Name}
-	}
+	exportNameFilter := "abcdefghijklmnopqrstuvwxyz1234567890"
 
+	for _, filter := range strings.Split(exportNameFilter, "") {
+		userPrompts, resp, err := proxy.getAllArchitectUserPrompts(ctx, true, true, filter+"*")
```

Comment:
Is this case insensitive? I mean will "a*" match any prompt that starts with "A"?

Reply:
Also, what about names that are not alphanumeric?

Reply:
Can we have names that are not alphanumeric?  Does prompts allow it?

Reply:
According to the API, names must be alphanumeric. It did not allow me to create a prompt with a non-alphanumeric name. However, I have seen some resources with non-alphanumeric names so i will add them to the check

Reply:
> Is this case insensitive? I mean will "a*" match any prompt that starts with "A"?

'a*' will return resources starting with both 'a' and 'A'. I will update the test case to confirm this

Reply:
> Can we have names that are not alphanumeric? Does prompts allow it?

On Further investigation, non-alphanumeric names are not valid, not needed to add them to the filter